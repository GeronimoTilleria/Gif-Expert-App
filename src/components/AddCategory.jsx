import { useState } from 'react';
import PropTypes from 'prop-types';

const initialInputValue = '';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState(initialInputValue);

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();

        const newValue = inputValue.trim();

        if (newValue.length <= 1) return;
        
        onNewCategory( newValue );
        setInputValue(initialInputValue);
    };

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input
                type='text'
                placeholder='Buscar Gifs'
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};

