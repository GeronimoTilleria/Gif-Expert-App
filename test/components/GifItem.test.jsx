import { GifItem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe('Prueba en <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://one-punch-man/saitama';

    test('debe de hacer el match en el snapshot', () => {
        const { container } = render(
            <GifItem
                title={ title }
                url={ url }
            />
        );

        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(
            <GifItem
                title={ title }
                url={ url }
            />
        );

        // screen.debug();
        // console.log( screen.getByRole('img') );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(
            <GifItem
                title={ title }
                url={ url }
            />
        );

        expect( screen.getByText( title ) ).toBeTruthy();
    });
});