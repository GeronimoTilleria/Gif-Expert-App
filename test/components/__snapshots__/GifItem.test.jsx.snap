// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> debe de hacer el match en el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch-man/saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
